packages/expo-modules-core/ios/Core/DynamicTypes/DynamicRawType.swift

```diff
@@ -19,6 +19,10 @@ internal struct DynamicRawType<InnerType>: AnyDynamicType {
     if let value = value as? InnerType {
       return value
     }
+    // Sometimes conversion from Double to Float will fail due to precision losses. We can accept them though.
+    if let value = value as? Double, wraps(Float.self) {
+      return Float(value)
+    }
```

Comment:
Actually I think this is no longer necessary – I recently introduced a separate dynamic type for numbers, see https://github.com/expo/expo/blob/main/packages/expo-modules-core/ios/Core/DynamicTypes/DynamicNumberType.swift